### ISSUE TITLE
The validator goes OOM while reading just 100 validators from the disk keystores

### ISSUE THREAD

**Issue by g11tech** [ChainSafe/lodestar#4166]
A discord user reported that validator goes OOM on startup while using the mev boost PR on ropsten network

![image](https://user-images.githubusercontent.com/76567250/174076133-27c40357-2db0-47c9-8b61-7836600d3520.png)

(https://discord.com/channels/593655374469660673/593655641445367808/986732729968246864)
However on debugging it was figured out that it was NOT related to mev boost PR and repro-able on the unstable branch. 
The validaor started without any issue on using mnemonic arguments. 

**Describe the bug**
Memory hog and OOM while reading keystores

The validator goes OOM while reading just `100` validators from  the disk keystores
**Expected behavior**
No unusual memory spikes and 

**Steps to Reproduce**
1. Import the keystore
2. Start the validaor

ubuntu docker,  lodestar Version: v0.39.0-dev.e5b74d5ae2

**Comment by dapplion** [comment]
If this issue is affecting all networks it's very critical, marked as `prio0-critical`

**Comment by dapplion** [comment]
Our beta testing process never runs a node with Keystores, so an issue like this won't be caught. We must run at least one node with mnemonics. Fox example, all the beta group nodes, run them with keystores since their key allocation is very stable

**Comment by g11tech** [comment]
> If this issue is affecting all networks it's very critical, marked as `prio0-critical`

yes i confirm it affects other networks as well: for keystores imported via deposit cli 100 vals, same behavior across various networks.


![image](https://user-images.githubusercontent.com/76567250/174349775-a9d74bcc-9f46-42ba-9b13-b98edf965125.png)
(the first process res memory climbed up till 7GB when screenshot, keeps going all the way up)

**Comment by g11tech** [comment]
UPDATE: cc @dapplion @wemeetagain 

have localized the problem over here: https://github.com/ChainSafe/lodestar/blob/unstable/packages/cli/src/validatorDir/ValidatorDir.ts#L114
this is being called in parallel for lets say 100 vals from here https://github.com/ChainSafe/lodestar/blob/unstable/packages/cli/src/validatorDir/ValidatorDirManager.ts#L75, and all promises get stuck here and memory starts increasing

if i restrict this to 10 vals, memory increase for lets say 2.5 GB and this resolves and we move forward, but the memory does NOT go down!

 It seems that "@chainsafe/bls-keystore"; is getting loaded into mem for each instance of `const privKey = await keystore.decrypt(password);` call 

If i serialize this call by removing promise.all from https://github.com/ChainSafe/lodestar/blob/unstable/packages/cli/src/validatorDir/ValidatorDirManager.ts#L75) then  issue resolves
but then its too slow to load 100 vals

**Comment by g11tech** [comment]
Opening it again for a proper fix the parallel keystore decrypt solution, the current one is too slow as also commented in the #4174 PR commit.
cc @dapplion @wemeetagain

**Comment by q9f** [comment]
Confirming the hotfix allows me to run 100 validators from keystore on Sepolia for now.

**Comment by g11tech** [comment]
JUST some more update regarding some further investigation done on this:

1. Repro-able in stable
2. Repro-able in v0.33.0 (docker), this is pre the crypto lib change, so this test removes the lib change from suspects
3. Memory doesn't come down (ran 20 vals  for 15 minutes which hogged 5gb), memory didn't drop

cc: @dapplion @wemeetagain

**Comment by dapplion** [comment]
Immediate issue is fixed, opened a follow-up issue to improve UX
- https://github.com/ChainSafe/lodestar/issues/4179
